integration-libs/opf/base/components/opf-error-modal/opf-error-modal.component.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostListener,
+  OnInit,
+} from '@angular/core';
+import { ErrorDialogOptions } from '@spartacus/opf/base/root';
+import { FocusConfig, LaunchDialogService } from '@spartacus/storefront';
+import { Observable, timer } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { OpfErrorModalService } from './opf-error-modal.service';
+
+@Component({
+  selector: 'cx-opf-error-modal',
+  templateUrl: './opf-error-modal.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OpfErrorModalComponent implements OnInit {
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'button',
+    focusOnEscape: true,
+  };
+
+  errorDialogOptions$: Observable<{ message: string; confirm: string }>;
+
+  @HostListener('click', ['$event'])
+  handleClick(event: UIEvent): void {
+    if ((event.target as any).tagName === this.el.nativeElement.tagName) {
+      this.dismissModal('Backdrop click');
+    }
+  }
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected el: ElementRef,
+    protected cd: ChangeDetectorRef,
+    protected opfErrorModalService: OpfErrorModalService
+  ) {
```

Comment:
[MEDIUM]

According to our [Coding Guidelines](https://wiki.one.int.sap/wiki/display/spar/Coding+Guidelines#CodingGuidelines-Useinject()functioninsteadofconstructorforDependencyInjection), in newly written source code, let's use `inject()` function instead of constructors for Dependency Injection.


You might find this automated conversion tool useful: https://ngxtension.netlify.app/utilities/migrations/inject-migration/